parts/k8s/addons/kubernetesmasteraddons-azure-npm-daemonset.yaml

```diff
@@ -76,7 +76,7 @@ spec:
         beta.kubernetes.io/os: linux
       containers:
         - name: azure-npm
-          image: mcr.microsoft.com/containernetworking/azure-npm:v1.0.18
+          image: {{ContainerImage "azure-npm-daemonset"}}
```

Comment:
Could you split this into two PRs? One for the defaults, which this PR was originally intended for, and one for the addon spec /ContainerImage refactor. This PR is really big now which makes it hard to properly review.

Reply:
I'll clean it up tomorrow, there's no easy way to split this up as the overall architecture debt means any piece meal change either introduces other breakage, or confusion. The intention here will be to rationalize all the various addons flows so that they work well together. I'll update the description and decorate the changeset to reflect that.